builtin/credential/jwt/path_login.go

```diff
@@ -194,11 +205,99 @@ func (b *jwtAuthBackend) pathLogin(ctx context.Context, req *logical.Request, d
 		return nil, err
 	}
 
+	if err := b.applyCelRoles(ctx, req, allClaims, auth); err != nil {
+		return nil, fmt.Errorf("error applying cel roles: %w", err)
+	}
+
 	return &logical.Response{
 		Auth: auth,
 	}, nil
 }
 
+func (b *jwtAuthBackend) applyCelRoles(ctx context.Context, req *logical.Request, allClaims map[string]any, auth *logical.Auth) error {
+	celRoleKeys, err := req.Storage.List(ctx, "cel/roles")
+	if err != nil {
+		return err
+	}
+	if len(celRoleKeys) == 0 {
+		return nil
+	}
+	for _, key := range celRoleKeys {
+		celRole, err := b.getCelRole(ctx, req.Storage, key)
```

Comment:
In particular, I don't quite like this behavior of evaluating all CEL policies against all tokens. It seems like it could lead to high complexity/latency if lots of CEL roles exist, which each validate a small portion of a token, rather than evaluating a single larger role.